pycalphad/model.py

```diff
@@ -986,23 +997,13 @@ def atomic_ordering_energy(self, dbe):
            This method overwrites the ``self.models`` dictionary with the model
            contributions for the disordered phase.
 
-        A peculiarity of this model is that the literature does not define how
-        to handle the energetic contributions from the sublattices that are
-        not participating in the disordering. There are no publications to our
-        knowledge that describe how to incorporate interstitial sublattices
-        into the model, whether to substitute their site fractions by mole
-        fractions like the substitutional phases or whether to leave the site
-        fractions as site fractions.
-
-        One could imagine how it might be of interest to model ordering on the
-        interstitial sublattices the same way that it is handled on the
-        substitional sublattices, but this has not been done as far as we are
-        aware. Therefore, we choose to treat only the first sublattice in the
-        disordered sublattice model as the substitional sublattice that may
-        disorder. All other sublattices are treated as interstitial sublattices
-        which are not substituted (i.e. they do not contribute to the ordering
-        energy). This formulation makes pycalphad conceptually consistent with
-        the implementation in OpenCalphad.
+        This method assumes that the first sublattice of the disordered phase is
+        the substitutional sublattice and all other sublattices are
+        interstitial. In the ordered phase, all sublattices with constituents
+        that match the disordered substitutional sublattice will be treated as
+        disordered (with site fractions replaced by mole fractions in the
+        ordered sublattices) and the interstitial sublattices will not have any
+        site fractions substituted.
```

Comment:
I don't know if I got it correctly, but for the ord-dis model, we should take the overall mole fraction and substitute in the ordered site fractions to get the ordering contribution, only if the phase can be totally disordered. But in the case of 
A2: (A,B,Va)m+n(C,Va)p
B2: (A,B,Va)m(A,B,Va)n(C,Va)p
The substitution shouldn't be done with mole fraction but with the sub-lattice fraction of the disordered state that represents the disordered state of the sub-lattice sets that undergo disordering of the ordered state. Is this what you mean in this phrase, am I missing something? e. g.:

Gm = Gm_dis + ΔGm_ord 
ΔGm_ord = Gm_ord (y1i, y2j, y3k) -  Gm_ord (y1i = y2j, y3k)

One reference that used this model: 
Acta Materialia 79 (2014) 1–15;
doi: http://dx.doi.org/10.1016/j.actamat.2014.07.006

Reply:
> The substitution shouldn't be done with mole fraction but with the sub-lattice fraction of the disordered state that represents the disordered state of the sub-lattice sets that undergo disordering of the ordered state.

Thank you for the reference. Most of the original publications for the partitioned order/disorder model do not include the interstitial sublattices and leave that out of their derivations. I actually had trouble tracking down a reference for how the interstitial sublattices are intended to work, so thank you for the recent reference.

### TL;DR

**I believe this formulation is the same as the one we are using, just with slightly different notation.** Please make any suggestions that would help clarify the documentation!

### Discussion

That Equation (11) from the work of Phan, Paek, and Kang, Acta Materialia 79 (2014) 1–15 that you cited:

<img width="479" alt="Screen Shot 2021-03-10 at 11 57 25 AM" src="https://user-images.githubusercontent.com/7681751/110666967-c92ac400-8197-11eb-95df-6e09be91054a.png">

treats the disordered sublattices the same as the one in Equation (3) of Servant and Ansara, Calphad 25, (2001) 509–525 that is cited in the documentation

<img width="529" alt="Screen Shot 2021-03-10 at 12 02 27 PM" src="https://user-images.githubusercontent.com/7681751/110667675-7e5d7c00-8198-11eb-8bf1-c1c2daae0372.png">

except that the work of Phan, Paek and Kang also specify that the interstitial sublattice should not be substituted (which is left out of Servant and Ansara's derivation, since they did not use an interstitial sublattice).

What the code here intends to do is as follows (you'll have to excuse the fast and loose notation here, since in pycalphad we don't make any assumptions about the number of sublattices so the notation from Phan, Paek and Kang which specifies the sublattices explicitly doesn't translate well):

<img width="1051" alt="image" src="https://user-images.githubusercontent.com/7681751/110668592-781bcf80-8199-11eb-8dbc-81f4461e0027.png">

where the "subs" superscript means "on substitutional sublattices" and the "inter" superscript means "on interstitial sublattices".  The "mole fraction" from `Model._mole_fraction` that is used here only is really more of a "quasi mole fraction" rather than a real "mole fraction". It's named in the spirit of the original derivations. `Model._mole_fraction` replaces the site fractions on substitutional sublattices and does not normalize the mole fraction for vacancy site fractions as a true "mole fraction" would. This is the motivation for making the `_mole_fraction` method private: because it's not a real mole fraction.

Reply:
For another brief description in the published literature, see section 3.3 of Connectable _et al._ Calphad 32 (2008) 361-370 [doi: 10.1016/j.calphad.2008.01.002](https://doi.org/10.1016/j.calphad.2008.01.002), specifically equations 14-18.

Reply:
Thank you @bocklund. This was really helpful. I think I could understand it better!